src/io/fasta.rs

```diff
@@ -22,11 +22,17 @@ use std::collections;
 use std::fs;
 use std::path::Path;
 use std::convert::AsRef;
+use std::cmp::min;
 
 use csv;
 
 use utils::{TextSlice, Text};
 
+
+/// Maximum size of temporary buffer used for reading indexed FASTA files.
+const MAX_FASTA_BUFFER_SIZE: usize = 512;
```

Comment:
What kind of pathological cases are you speaking about that make this necessary? No objection against it, I just want to understand.